presto-main/src/main/java/com/facebook/presto/operator/HashBuilderOperator.java

```diff
@@ -629,10 +636,13 @@ private SingleStreamSpiller getSpiller()
     @Override
     public void close()
     {
+        if (state == State.CLOSED) {
+            return;
```

Comment:
Could we have a test or some other measure to verify close is idempotent? 
Maybe at least some hack in `com.facebook.presto.testing.LocalQueryRunner#executeInternal` ?

Reply:
`Driver` has extensive checks that prevent close from being called multiple times, so the only way to test this is to expose operators directly outside of `Driver`.  I'm reluctant to do this given the threading model of drivers, but we could for this test.

I think the better approach in the long run is to wrapper every operator to simplify the state machine (and may other items).

Reply:
In tests, we sometimes have operators without a Driver. e.g vide `OperatorAssertion`.

> I think the better approach in the long run is to wrapper every operator to simplify the state machine (and may other items).

This is a fine idea. However, LJO calls close on itself. Even if wrapped, it would still cause problems (unless the wrapped is made aware of wrapping).